File: An2/Sem1/MAP/MAP-Java/Partial/src/Controller/Controller.java
package Controller;

import Model.Cires;
import Model.Mar;
import Model.Par;
import Model.Pom;
import Repository.Repository;
import Utils.MyException;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;

/**
 * Created with IntelliJ IDEA.
 * User: mihai
 * Date: 12/16/13
 * Time: 10:22 AM
 * To change this template use File | Settings | File Templates.
 */
public class Controller {

    private Repository repo;

    public Controller(Repository repo) {
        this.repo = repo;
        try {
            this.readFromFile("input.txt");
            this.writeToFile(this.whatToWrite());
        } catch (MyException e) {
            e.printStackTrace();  //To change body of catch statement use File | Settings | File Templates.
        }
    }

    public void writeToFile(ArrayList<Pom> toWrite) throws MyException {
        try{
            if (toWrite.size() < 0)
                throw new MyException("No elements to write!");
            repo.writeToFile(toWrite);
        }catch (MyException e){
            System.out.println(e.getMessage());
        }
    }

    public ArrayList<Pom> whatToWrite(){
        try{
            ArrayList<Pom> copy = new ArrayList<Pom>(repo.getElements());
            ArrayList<Pom> result = new ArrayList<Pom>();
            for (Pom elem : copy){
                if ((!"Cires".equals(elem.getMyClass())) && (elem.getAge() > 3))
                    result.add(elem);
            }
            return result;
        }catch (NullPointerException e){
            System.out.println(e.getMessage());
        }
        return null;
    }

    public void readFromFile(String fileName){
        try{
            ArrayList<Pom> newElements = new ArrayList<Pom>();
            BufferedReader br = new BufferedReader(new FileReader(fileName));
            String line;
            while ((line = br.readLine()) != null){
                String[] tokens = line.split(" ");
                if ("Cires".equals(tokens[0]))
                    newElements.add(new Cires(tokens[1],Integer.parseInt(tokens[2])));
                else if ("Mar".equals(tokens[0]))
                    newElements.add(new Mar(tokens[1],Integer.parseInt(tokens[2])));
                else if ("Par".equals(tokens[0]))
                    newElements.add(new Par(tokens[1],Integer.parseInt(tokens[2])));
                else
                    throw new MyException("Error reading from file!");

                this.repo.replaceContent(newElements);
            }
        }catch (IOException e){
            System.out.println(e.getMessage());
        }catch (MyException e) {
            System.out.println(e.getMessage());
        }catch (NumberFormatException e){
            System.out.println(e.getMessage());
        }
    }

}

File: An2/Sem1/MAP/MAP-C#/Partial/Partial/Controller/Controller.cs
using Partial.Model;
using Partial.Utils;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Partial.Controller
{
    class Controller
    {
        private Repository.RepoInterface<Leguma> repo;

        public Controller(Repository.RepoInterface<Leguma> repo)
        {
            this.repo = repo;
            this.readFromFile("input.txt");
            this.writeToFile(this.whatToWrite());
        }

        public void readFromFile(String fileName)
        {
            try
            {
                LinkedList<Leguma> newList = new LinkedList<Leguma>();
                StreamReader sr = new StreamReader(fileName);
                String line;
                String[] tokens;

                while ((line = sr.ReadLine()) != null)
                {
                    tokens = line.Split(new String[] { " " }, StringSplitOptions.None);
                    if ("Rosie".Equals(tokens[0]))
                        newList.AddLast(new Rosie(tokens[1], float.Parse(tokens[2])));
                    else if ("Ardei".Equals(tokens[0]))
                        newList.AddLast(new Ardei(tokens[1], float.Parse(tokens[2])));
                    else if ("Vinete".Equals(tokens[0]))
                        newList.AddLast(new Vinete(tokens[1], float.Parse(tokens[2])));
                    else throw new MyException("Error reading from file!");

                }

                this.repo.replaceContent(newList);

            }
            catch (IOException e)
            {

            }
            catch (MyException e)
            {
            }
            catch (FormatException)
            {

            }

        }

        public void writeToFile(LinkedList<Leguma> toWrite)
        {
            try
            {
                if (toWrite.Count() < 0)
                    throw new MyException("Nothing to write!");
                repo.writeToFile(toWrite);
            }
            catch (IOException)
            {

            }
            catch (MyException)
            {

            }
        }

        public LinkedList<Leguma> whatToWrite()
        {
            try
            {
                LinkedList<Leguma> copy = new LinkedList<Leguma>(repo.getElements());
                LinkedList<Leguma> result = new LinkedList<Leguma>();

                foreach (Leguma elem in copy){
                    if ("Vinete".Equals(elem.getMyClass()) && (elem.getWeight() > 0.2))
                    {
                        result.AddLast(elem);
                    }
                }

                return result;

            }
            catch (NullReferenceException)
            {

            }

            return null;

        }

    }
}
